src/PureScript/Backend/Optimizer/Codegen/Tco.purs

```diff
@@ -295,12 +295,9 @@ analyze env (NeutralExpr expr) = case expr of
     TcoExpr analysis $ PrimEffect (EffectRefWrite ref' val')
   Branch branches def -> do
     let branches' = map (\(Pair a b) -> Pair (overTcoAnalysis tcoNoTailCalls (analyze env a)) (analyze env b)) branches
-    let def' = analyze env <$> def
-    let analysis@(TcoAnalysis { total }) = foldMap (foldMap tcoAnalysisOf) branches' <> foldMap tcoAnalysisOf def'
-    if isJust def then
-      TcoExpr (tcoTotal (Root total) analysis) $ Branch branches' def'
-    else
-      TcoExpr (tcoTotal Partial analysis) $ Branch branches' def'
+    let def' = analyze env def
+    let analysis@(TcoAnalysis { total }) = foldMap (foldMap tcoAnalysisOf) branches' <> tcoAnalysisOf def'
+    TcoExpr (tcoTotal (Root total) analysis) $ Branch branches' def'
```

Comment:
Since we are removing the `Partial` case here, is it referenced anywhere else? I think this particular analysis is unnecessary without backtracking.